File: app/src/main/java/org/neidhardt/dynamicsoundboard/views/BaseConfirmDeleteDialog.java
package org.neidhardt.dynamicsoundboard.views;

import android.annotation.SuppressLint;
import android.app.Dialog;
import android.os.Bundle;
import android.support.v7.app.AppCompatDialog;
import android.view.View;
import android.widget.TextView;
import org.neidhardt.dynamicsoundboard.R;

/**
 * File created by eric.neidhardt on 16.02.2015.
 */
public abstract class BaseConfirmDeleteDialog extends BaseDialog implements View.OnClickListener
{
	@Override
	public Dialog onCreateDialog(Bundle savedInstanceState)
	{
		@SuppressLint("InflateParams") View view = this.getActivity().getLayoutInflater().inflate(R.layout.dialog_confirm_delete, null);
		TextView infoText = (TextView) view.findViewById(R.id.tv_message);
		infoText.setText(this.getInfoTextResource());

		view.findViewById(R.id.b_cancel).setOnClickListener(this);
		view.findViewById(R.id.b_ok).setOnClickListener(this);

		AppCompatDialog dialog = new AppCompatDialog(this.getActivity(), R.style.DialogThemeNoTitle);
		dialog.setContentView(view);
		return dialog;
	}

	@Override
	public void onClick(View v)
	{
		switch (v.getId())
		{
			case R.id.b_cancel:
				this.dismiss();
				break;
			case R.id.b_ok:
				this.delete();
				this.dismiss();
		}
	}

	protected abstract int getInfoTextResource();

	protected abstract void delete();
}

File: app/src/main/kotlin/org/neidhardt/dynamicsoundboard/views/BaseConfirmDeleteDialog.kt
package org.neidhardt.dynamicsoundboard.views

import android.annotation.SuppressLint
import android.app.Dialog
import android.os.Bundle
import android.support.v7.app.AlertDialog
import android.widget.TextView
import org.neidhardt.dynamicsoundboard.R

/**
 * File created by eric.neidhardt on 16.02.2015.
 */
abstract class BaseConfirmDeleteDialog : BaseDialog()
{

	override fun onCreateDialog(savedInstanceState: Bundle?): Dialog
	{
		@SuppressLint("InflateParams") val view = this.activity.layoutInflater.inflate(R.layout.dialog_confirm_delete, null)
		val infoText = view.findViewById(R.id.tv_message) as TextView
		infoText.setText(this.infoTextResource)

		return AlertDialog.Builder(this.activity).apply {
			this.setView(view)
			this.setNegativeButton(R.string.dialog_cancel, { dialogInterface, i -> dismiss() })
			this.setPositiveButton(R.string.dialog_delete, { dialogInterface, i ->
				delete()
				dismiss() })
		}.create()
	}

	protected abstract val infoTextResource: Int

	protected abstract fun delete()
}
